packages/docusaurus-plugin-content-blog/src/blogUtils.ts

```diff
@@ -44,15 +43,40 @@ export function getSourceToPermalink(
   );
 }
 
-// YYYY-MM-DD-{name}.mdx?
-// Prefer named capture, but older Node versions do not support it.
-const DATE_FILENAME_PATTERN = /^(\d{4}-\d{1,2}-\d{1,2})-?(.*?).mdx?$/;
-
-function toUrl({date, link}: DateLink) {
-  return `${date
-    .toISOString()
-    .substring(0, '2019-01-01'.length)
-    .replace(/-/g, '/')}/${link}`;
+const DATE_FILENAME_REGEX = /^(?<date>\d{4}[-\/]\d{1,2}[-\/]\d{1,2})[-\/]?(?<text>.*?)(\/index)?.mdx?$/;
+
+type ParsedBlogFileName = {
+  date: Date | undefined;
+  text: string;
+  slug: string;
+};
+
+export function parseBlogFileName(
+  blogSourceRelative: string,
+): ParsedBlogFileName {
+  const dateFilenameMatch = blogSourceRelative.match(DATE_FILENAME_REGEX);
+  if (dateFilenameMatch) {
+    const dateString = dateFilenameMatch.groups!.date!;
+    const text = dateFilenameMatch.groups!.text!;
+    // Always treat dates as UTC by adding the `Z`
+    const date = new Date(`${dateString}Z`);
+    // TODO use replaceAll once we require NodeJS 16
```

Comment:
Using `replaceAll` is probably safe because the compilation target is set to `ES2019` anyways. Only the compiled JS file is actually used, and the TS syntax doesn't matter that much

Reply:
The compiled output remains "replaceAll" so this won't work in NodeJS versions we support unfortunately. 

Will keep current code but if you find a way to refactor this and make sure we can use newer apis I'd be happy to have a solution. Afaik the target only make sure NodeJS can parse the syntax, I'm not sure it will replace/polyfill the apis that are not supported by this NodeJS version.

Reply:
Not understand, why not just use regex (`dateString.replace(/-/g, '/')`)?

Reply:
yes that would also work 😅